### ISSUE TITLE
AST Reconstruction Loss Function

### ISSUE THREAD

**Issue by timlawrenz** [timlawrenz/jubilant-palm-tree#35]
Description: Implement a loss function in src/loss.py that can measure the difference between the original AST and the reconstructed AST. This is not a simple MSE.

Approach: The loss will likely be a combination of:

    Node Type Loss: A cross-entropy loss for predicting the correct type of each node.

    Edge Prediction Loss: A loss function (e.g., binary cross-entropy) for predicting the correct parent-child connections between nodes.

Definition of Done:

    A custom loss function is implemented that can compare two torch_geometric.data.Data objects.

    The loss function returns a single scalar value.

After that, update the README to reflect those changes
